docs/provider-pkcs11.7.md

```diff
@@ -0,0 +1,265 @@
+% provider-pkcs11(7) | Configuration directives
+
+NAME
+====
+pkcs11-provider - An OpenSSL provider that allows to directly interface
+with pkcs11 drivers.
+
+
+DESCRIPTION
+===========
+
+Starting with version 3.0 the OpenSSL project introduced a new modular
+system to extend OpenSSL that replaces the deprecated Engines modules.
+
+Providers[1] are loaded via configuration directives in the openssl
+configuration file (or directly loaded by applications).
+
+The pkcs11 provider allows applications linked to openssl to use keys
+and cryptographic operations from a hardware or software token via
+their PKCS#11[2] driver and the use of pkcs11 URIs[3].
+
+The pkcs11 provider can be configured to be automatically loaded via
+openssl.cnf
+
+
+CONFIGURATION
+=============
+Configuration options recognized by the provider
+
+## pkcs11-module-path
+
+A file path to the pkcs11 driver to be used
+
+Default: If compiled with p11-kit defaults to its driver, otherwise none.
```

Comment:
I would probably clarify what p11-kit driver as the p11-kit-proxy is the default, which should contain any pkcs11 module in the system. Something along the lines 
```suggestion
Default: If compiled with p11-kit defaults to its proxy driver, otherwise none.
```
Unfortunately I do not see any useful manual page describing this except the manual online so maybe linking this might work.

https://p11-glue.github.io/p11-glue/p11-kit/manual/sharing.html